src/main/java/ua/softserveinc/tc/validator/RoomValidatorImpl.java

```diff
@@ -96,26 +102,36 @@ public void validate(Object o, Errors errors) {
                 errors.rejectValue(ValidationConstants.MANAGERS_FIELD, ValidationConstants.ROOM_MANAGER_INVALID);
             } else {
                 List<UserDto> managers = Arrays.asList(new Gson().fromJson(jsonManagers, UserDto[].class));
-                List<Long> managerId = new ArrayList<>();
-                for (UserDto id : managers) {
-                    managerId.add(id.getId());
-                }
+                List<Long> managerId = JsonUtil.fromJsonList(roomToValidate.getManagers(), UserDto[].class).stream()
+                        .map(UserDto::getId).collect(Collectors.toList());
                 if (managers.stream().anyMatch(manager -> (manager.getId() == null))) {
                     errors.rejectValue(ValidationConstants.MANAGERS_FIELD, ValidationConstants.ROOM_MANAGER_EMPTY);
-                }
-                if (managerId.stream().distinct().count() != managerId.size()) {
-                    errors.rejectValue(ValidationConstants.MANAGERS_FIELD, ValidationConstants.ROOM_MANAGER_DUPLICATE);
+                } else {
+                    if (managerId.stream().distinct().count() != managerId.size()) {
+                        errors.rejectValue(ValidationConstants.MANAGERS_FIELD, ValidationConstants.ROOM_MANAGER_DUPLICATE);
+                    }
+                    for(Long id: managerId) {
+                        if (userService.findByIdTransactional(id).getRole() != Role.MANAGER) {
```

Comment:
It is very bad practice to send requests to database in loop. It is mush better do that in batch.